### PR TITLE
Feature6

### PR DESCRIPTION
**Pull request by dsteinberg**
Implements bootstrap cross evaluation.

I feel like we could simplify the `groups` interface in this and in `bootstrap_model` interfaces - or at least detect when an estimator can accept groups and automatically use them - and warn a user when not using groups and `GridSearchCV` is used?

### CODE REVIEW

**Comment by jack-fs** [comment]
Agreed -- it would be nice to have a smoother  way to use the groups. This looks good to me!

File: cinspect/model_evaluation.py [hunk 1]
```diff
@@ -35,32 +36,27 @@ def crossval_model(
...
-        if hasattr(X, "iloc"):
-            Xs, Xr = X.iloc[sind], X.iloc[rind]
-        else:
-            Xs, Xr = X[sind], X[rind]
-        if hasattr(y, "iloc"):
-            ys, yr = y.iloc[sind], y.iloc[rind]
-        else:
-            ys, yr = y[sind], y[rind]
+
+        Xr, Xs = _split_data(X, rind, sind)
```
Comment (jack-fs) [inline cinspect/model_evaluation.py:54]: Nice to have this factored out

File: tests/test_model_evaluation.py [hunk 2]
```diff
@@ -230,6 +233,24 @@ def sample_from_normal(rng):
...
+def test_bootcross_split(random_state, test_size):
+    """Make sure the bootstrap - splitting is working as intended."""
+    N = 1000
+    random_state = check_random_state(random_state)
+    tri, tsi = _bootcross_split(N, test_size, random_state)
+    
+    # Test size of test set
+    assert len(tsi) == test_size
+
+    # Make sure training and testing are not overlapping
```
Comment (jack-fs) [inline tests/test_model_evaluation.py:247]: checking emptiness of `set.intersection()` might be more efficient... insignificant in the scheme of things though

**Review by jack-fs — APPROVED** [review]
Looks good!

File: tests/test_model_evaluation.py (hunk 2 above)
Comment (dsteinberg) [reply tests/test_model_evaluation.py:247]: Good point! I'll put this in for my next PR

**Pull request merged**